src/outputgeneration/ParseTreeWriter.js

```diff
@@ -1006,9 +1006,12 @@ export class ParseTreeWriter extends ParseTreeVisitor {
    * @param {NewExpression} tree
    */
   visitNewExpression(tree) {
+    var complexOperand = tree.operand.operand !== void 0;
```

Comment:
Right now, it's relying on the vm's own idea of precedence, but I think the parser is mostly getting it right, so always parenthesizing the operand should be fine (just produces some ugly code).

Maybe it makes more sense to cover "complex" operands in transformNewExpression()